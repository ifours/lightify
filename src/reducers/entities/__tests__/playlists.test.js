import reducer, { initialState } from '../playlists';

describe('Playlists entites reducer', () => {
  it('should return initialState', () => {
    const state = reducer(initialState, {});

    expect(state).toMatchSnapshot();
    expect(state).toBe(initialState);
  });

  it('should return the same state if payload doesn\'t have albums entites', () => {
    const state = { foo: 'bar' };
    const newState = reducer(state, { type: 'quz', payload: { foo: 'bar' } });

    expect(newState).toMatchSnapshot();
    expect(newState).toBe(state);
  });

  it('should return new state if payload has albums entites', () => {
    const state = { foo: 'bar' };
    const payload = { entities: { playlists: { quz: 'baz' } } };
    const newState = reducer(state, { type: 'quz', payload });

    expect(newState).toMatchSnapshot();
    expect(newState).not.toBe(state);
  });

  it('should return new state', () => {
    const state = { foo: 'bar' };
    const payload = { entities: { playlists: { quz: 'baz', foo: 'new value' } } };
    const newState = reducer(state, { type: 'quz', payload });

    expect(newState).toMatchSnapshot();
    expect(newState).not.toBe(state);
  });

  it('should return new state and update playlist\'s tracks', () => {
    const playlistId = 'fooId';
    const state = { [playlistId]: { tracks: { foo: 'bar' } } };
    const payload = {
      entities: { playlists: { quz: 'baz', foo: 'new value' } },
      result: ['foo', 'bar', 'bas'],
      playlistId
    };
    const newState = reducer(state, { type: 'FETCH_TRACKS_SUCCEEDED', payload });

    expect(newState).toMatchSnapshot();
    expect(newState).not.toBe(state);
  });
});
