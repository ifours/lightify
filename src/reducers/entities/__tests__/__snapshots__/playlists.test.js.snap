// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlists entites reducer should return initialState 1`] = `Object {}`;

exports[`Playlists entites reducer should return new state 1`] = `
Object {
  "foo": "new value",
  "quz": "baz",
}
`;

exports[`Playlists entites reducer should return new state and update playlist's tracks 1`] = `
Object {
  "fooId": Object {
    "tracks": Object {
      "foo": "bar",
      "items": Array [
        "foo",
        "bar",
        "bas",
      ],
    },
  },
}
`;

exports[`Playlists entites reducer should return new state if payload has albums entites 1`] = `
Object {
  "foo": "bar",
  "quz": "baz",
}
`;

exports[`Playlists entites reducer should return the same state if payload doesn't have albums entites 1`] = `
Object {
  "foo": "bar",
}
`;
